flowey/flowey_lib_hvlite/src/_jobs/check_openhcl_size.rs

```diff
@@ -0,0 +1,119 @@
+        } = request;
+
+        let repo_path = ctx.reqv(crate::git_checkout_openvmm_repo::req::GetRepoDir);
+        let xtask = ctx.reqv(|v| crate::build_xtask::Request {
+            target: xtask_target,
+            xtask: v,
+        });
+        let openhcl_vmm = ctx.reqv(|v| crate::build_openvmm_hcl::Request {
+            build_params: crate::build_openvmm_hcl::OpenvmmHclBuildParams {
+                target: openvmm_hcl_target,
+                profile: crate::build_openvmm_hcl::OpenvmmHclBuildProfile::OpenvmmHclShip,
+                features: [OpenvmmHclFeature::Tpm].into(),
+                no_split_dbg_info: false,
+            },
+            openvmm_hcl_output: v,
+        });
+
+        let did_verify_usermode_size =
+            ctx.emit_rust_step("verify openhcl_vmm usermode bin size", |ctx| {
+                let xtask = xtask.clone().claim(ctx);
+                let repo_path = repo_path.clone().claim(ctx);
+                let openhcl_vmm = openhcl_vmm.claim(ctx);
+
+                |rt| {
+                    let xtask = match rt.read(xtask) {
+                        crate::build_xtask::XtaskOutput::LinuxBin { bin, .. } => bin,
+                        crate::build_xtask::XtaskOutput::WindowsBin { exe, .. } => exe,
+                    };
+                    let openhcl_vmm = rt.read(openhcl_vmm).bin;
+                    let sh = xshell::Shell::new()?;
+                    sh.change_dir(rt.read(repo_path));
+                    xshell::cmd!(
+                        sh,
+                        "{xtask} verify-size
+                        -t underhill-x86_64-unknown-linux-musl-release
+                        -p {openhcl_vmm}
+                    "
+                    )
+                    .run()?;
+                    Ok(())
+                }
+            });
+
+        let openhcl_kernel =
+            ctx.reqv(
+                |v| crate::download_openhcl_kernel_package::Request::GetPackage {
+                    kind: OpenhclKernelPackageKind::Main,
+                    arch: OpenhclKernelPackageArch::X86_64,
+                    pkg: v,
+                },
+            );
+
+        let did_verify_kernel_size = ctx.emit_rust_step("verify OpenHCL kernel size", |ctx| {
+            let xtask = xtask.claim(ctx);
+            let repo_path = repo_path.claim(ctx);
+            let openhcl_kernel = openhcl_kernel.claim(ctx);
+            |rt| {
+                let xtask = match rt.read(xtask) {
+                    crate::build_xtask::XtaskOutput::LinuxBin { bin, .. } => bin,
+                    crate::build_xtask::XtaskOutput::WindowsBin { exe, .. } => exe,
+                };
+                let vmlinux = rt
+                    .read(openhcl_kernel)
+                    .join("build/native/bin/x64/vmlinux.dbg");
+                let sh = xshell::Shell::new()?;
+                sh.change_dir(rt.read(repo_path));
+                xshell::cmd!(
+                    sh,
+                    "{xtask} verify-size
```

Comment:
In a future change we might want to consider creating baselines for the ARM64 kernel (and probably the CVM kernel as well). But that can be done in a future change.

Reply:
Yeah definitely

Reply:
We also probably want to look into having the task download builds from other steps, rather than rebuilding it itself.

Reply:
yeah good call.